genesyscloud/flow_loglevel/genesyscloud_flow_loglevel_proxy.go

```diff
@@ -0,0 +1,164 @@
+	return p.createFlowLogLevelAttr(ctx, p, flowLogLevelId, flowLogLevelRequest)
+}
+
+// getFlowLogLevelById returns a single Genesys Cloud External Contact by Id
+func (p *flowLogLevelProxy) getFlowLogLevelById(ctx context.Context, flowId string) (*platformclientv2.Flowsettingsresponse, *platformclientv2.APIResponse, error) {
+	return p.getFlowLogLevelByIdAttr(ctx, p, flowId)
+}
+
+// updateFlowLogLevel updates a Genesys Cloud External Contact
+func (p *flowLogLevelProxy) updateFlowLogLevel(ctx context.Context, flowId string, flowLogLevelRequest *platformclientv2.Flowloglevelrequest) (*platformclientv2.Flowsettingsresponse, *platformclientv2.APIResponse, error) {
+	return p.updateFlowLogLevelAttr(ctx, p, flowId, flowLogLevelRequest)
+}
+
+// DeleteFlowLogLevel deletes a Genesys Cloud External Contact by Id
+func (p *flowLogLevelProxy) deleteFlowLogLevelById(ctx context.Context, flowId string) (*platformclientv2.APIResponse, error) {
+	return p.deleteFlowLogLevelByIdAttr(ctx, p, flowId)
+}
+
+// createFlowLogLevelFn is an implementation function for creating a Genesys Cloud External Contact
+func createFlowLogLevelFn(ctx context.Context, p *flowLogLevelProxy, flowId string, flowLogLevelRequest *platformclientv2.Flowloglevelrequest) (*platformclientv2.Flowsettingsresponse, *platformclientv2.APIResponse, error) {
+	flowLogLevel, apiResponse, err := p.architectApi.PostFlowInstancesSettingsLoglevels(flowId, *flowLogLevelRequest, []string{"logLevelCharacteristics.characteristics"})
+
+	if err != nil {
+		return nil, apiResponse, fmt.Errorf("Failed to create flow log level: %s", err)
+	}
+
+	return flowLogLevel, apiResponse, nil
+}
+
+// getFlowLogLevelByIdFn is an implementation of the function to get a Genesys Cloud External Contact by Id
+func getFlowLogLevelByIdFn(ctx context.Context, p *flowLogLevelProxy, flowId string) (*platformclientv2.Flowsettingsresponse, *platformclientv2.APIResponse, error) {
+	flowLogLevel, apiResponse, err := p.architectApi.GetFlowInstancesSettingsLoglevels(flowId, []string{"logLevelCharacteristics.characteristics"})
+	if err != nil {
+		return nil, apiResponse, fmt.Errorf("Failed to retrieve flow log level by id %s: %s", flowId, err)
+	}
+
+	return flowLogLevel, apiResponse, nil
+}
+
+// getAllFlowLogLevelsFn is the implementation for retrieving all flow log levels in Genesys Cloud
+func getAllFlowLogLevelsFn(ctx context.Context, p *flowLogLevelProxy) (*[]platformclientv2.Flowsettingsresponse, *platformclientv2.APIResponse, error) {
+	const pageSize = 100
+	var totalFlowLogLevels []platformclientv2.Flowsettingsresponse
+
+	flowSettingsResponse, apiResponse, err := p.architectApi.GetFlowsInstancesSettingsLoglevels([]string{"logLevelCharacteristics.characteristics"}, 1, pageSize)
+	if err != nil {
+		return nil, apiResponse, fmt.Errorf("Failed to get page of flows: %v", err)
+	}
+
+	for _, flowLogLevel := range *flowSettingsResponse.Entities {
```

Comment:
We are using a new syntax when looping through an array and adding it to another array. You can do this instead
`totalFlowLogLevels = append(totalFlowLogLevels, *flowSettingsResponse.Entities...)`